scripts/find-ts-version.js

```diff
@@ -1,8 +1,8 @@
 // @ts-check
 const fs = require("fs");
 
-const FIRST_TYPESCRIPT_LINE = /typescript\@\^?(\d+.\d+(.\d+)?):/;
-const TYPESCRIPT_VERSION_LINE = /version "(\d+.\d+)(.\d+)?"/;
+const FIRST_TYPESCRIPT_LINE = /typescript@\S*:/;
```

Comment:
I see that it may be useful for beta and preview versions of TypeScript. But currently, we don't use those, so it's unclear why this change is important. Would you be able to clarify your reasons?

Reply:
The current regex only matches versions prefixed with `^`, but when testing locally, I often change the version to something like `~4.5` to get the latest patch of a minor version. However, this causes `yarn test` to fail due to the regex.

Also, since we're not using any of the captured groups from this regex, it felt like the regex could be simplified. Let me know if you think this approach might produce false positives. Maybe I should update the regex to this:

```suggestion
const FIRST_TYPESCRIPT_LINE = /^typescript@\S*:$/;
```

Reply:
@som-sm good point on `~4.5`, I'm happy to keep it